src/mame/heathkit/tlb.cpp

```diff
@@ -305,38 +305,57 @@ void heath_tlb_device::right_shift_w(int state)
 
 MC6845_UPDATE_ROW(heath_tlb_device::crtc_update_row)
 {
-	if (!de)
-	{
-		return;
-	}
-
 	rgb_t const *const palette = m_palette->palette()->entry_list_raw();
 	uint32_t *p = &bitmap.pix(y);
 
-	for (uint16_t x = 0; x < x_count; x++)
-	{
-		uint8_t inv = (x == cursor_x) ? 0xff : 0;
-
-		uint8_t chr = m_p_videoram[(ma + x) & 0x7ff];
+	if (de) {
 
-		if (chr & 0x80)
+		for (uint16_t x = 0; x < x_count; x++)
 		{
-			inv ^= 0xff;
-			chr &= 0x7f;
+			uint8_t inv = (x == cursor_x) ? 0xff : 0;
+
+			uint8_t chr = m_p_videoram[(ma + x) & 0x7ff];
+
+			if (chr & 0x80)
+			{
+				inv ^= 0xff;
+				chr &= 0x7f;
+			}
+
+			// get pattern of pixels for that character scanline
+			uint8_t const gfx = m_p_chargen[(chr<<4) | ra] ^ inv;
+
+			// Display a scanline of a character (8 pixels)
+			*p++ = palette[BIT(gfx, 7)];
+			*p++ = palette[BIT(gfx, 6)];
+			*p++ = palette[BIT(gfx, 5)];
+			*p++ = palette[BIT(gfx, 4)];
+			*p++ = palette[BIT(gfx, 3)];
+			*p++ = palette[BIT(gfx, 2)];
+			*p++ = palette[BIT(gfx, 1)];
+			*p++ = palette[BIT(gfx, 0)];
+		}
+	}
+	else
+	{
+		// for the additional scan lines we need to clear them to properly support
+		// line 25 of the H19. On the real H19, the crtc is reprogramed to not
+		// display these lines and the memory remains the same. Only when the line
+		// 25 is reenabled will the memory be cleared. On a real CRT the screen
+		// will fade, but here, have to explicitly clear them from the screen.
```

Comment:
This comment seems to misunderstand how emulated video works.  The system is drawing black if DE is inactive, so you need to fill the relevant part of the bitmap with black.  The output bitmap represents beam intensity.  It isn’t automatically cleared between frames.  If the system outputs black, you need to fill the applicable parts of the output bitmap with black.  Phosphor persistence, beam focus, video bandwidth limitations, etc. is handled in shaders.

Reply:
I've removed the comment, but thinking about it, it seems since screen_device represents the beam intensity, it should start each scan out with an zeroed array.

Reply:
It doesn’t because that would be additional overhead when most systems are going to be rendering complete lines and overwriting it anyway.  There’s no reason to fill it before telling the system to update the dirty area.

Also, there’s more than one kind of raster bitmap display.  Consider a passive matrix LCD (e.g. the Nintendo Game Boy).  In that case, the undriven state is “whiter than white”, and driven pixels are darkened.

TL;DR the system/device is responsible for filling the area it’s told to update, and must not make assumptions about what it contains on entry.